src/modm/platform/core/cortex/module.lb

```diff
@@ -73,11 +75,34 @@ def common_memories(env):
     """
     device = env[":target"]
     memories = listify(device.get_driver("core")["memory"])
+
+    # Convert from string to int and add offsets
+    flash_offset = env.get_option(":platform:cortex-m:linkerscript.flash_offset", 0)
+    for m in memories:
+        if m["name"] == "flash" and flash_offset != 0:
+            # The offset needs to be aligned regarding the number of interrupts
+            # in order for the vector table relocation to work. Every interrupt
+            # requires 4 bytes, and therefore two additional bits of alignment.
+            number_of_interrupts = env.query(":::vector_table")["highest_irq"] + 16
+            bit_alignment = number_of_interrupts.bit_length() + 2
+            mask = (1 << bit_alignment) - 1
+            if flash_offset & mask:
+                raise ValidateException("Invalid flash offset in option '{}'. "
+                                        "The offset needs to be {} bit aligned."
+                                        .format("linkerscript.flash_offset",
+                                                bit_alignment))
```

Comment:
You may want to add a validate function and call the query there, so that this is raised in the right context. This is enough:
```py
def validate(env):
    env.query("vector_table")
```

Reply:
I've moved the complete validation to the `validate` step.